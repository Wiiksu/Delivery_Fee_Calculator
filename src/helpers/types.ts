import { InputHTMLAttributes } from "react";

export interface InputFieldProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  errorMessage?: string;
}

export type ButtonPropTypes = {
  title: string;
  type: "submit" | "reset";
};

export type UserFormType = {
  cartValue: number;
  deliveryDistance: number;
  numberOfItems: number;
  orderTime: Date;
};

export type UserFormReturnType = {
  cartValueSurcharge: number;
  deliveryDistanceSurcharge: number;
  numberOfItemsSurcharge: number;
  orderTimeSurcharge: number;
  orderTotal: number;
  freeOrder: boolean;
};

export interface UserFormProps {
  saveCalculationResults: (results: UserFormReturnType | undefined) => void;
}

export interface ResultProps {
  orderResults?: UserFormReturnType;
}

type SubInfoType = {
  title: string;
  info: string;
};

export type ItemsType = {
  title: string;
  subinformation: SubInfoType[];
};
