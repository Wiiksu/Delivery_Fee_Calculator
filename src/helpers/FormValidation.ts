import * as yup from "yup";

export const formValidatorSchema = yup.object().shape({
  cartValue: yup
    .number()
    .transform((value, originalValue) =>
      String(originalValue).trim() === "" ? null : value
    )
    .typeError("Value must be a number")
    .positive("Can't be a negative number")
    .required("Cart value is required"),
  deliveryDistance: yup
    .number()
    .transform((value, originalValue) =>
      String(originalValue).trim() === "" ? null : value
    )
    .typeError("Distance must be a number")
    .positive("Can't be a negative number")
    .required("Delivery distance is required"),
  numberOfItems: yup
    .number()
    .transform((value, originalValue) =>
      String(originalValue).trim() === "" ? null : value
    )
    .typeError("Amount of items must be a number")
    .positive("Can't be a negative number")
    .required("Amount of items is required"),
  orderTime: yup
    .date()
    .transform(function (value) {
      return this.isType(value) ? value : undefined;
    })
    .required("Valid time is required"),
});
