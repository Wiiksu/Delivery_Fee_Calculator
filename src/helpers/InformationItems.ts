import { ItemsType } from "./types";

// List of items for the information-component mapping
export const items: ItemsType[] = [
  {
    title: "About this app",
    subinformation: [
      {
        title: "Cart Value",
        info: "If the cart value is less than 10€, a small order surcharge is added to the delivery price. The surcharge is the difference between the cart value and 10€. For example if the cart value is 8.90€, the surcharge will be 1.10€.",
      },
      {
        title: "Delivery Distance",
        info: "A delivery fee for the first 1000 meters (=1km) is 2€. If the delivery distance is longer than that, 1€ is added for every additional 500 meters that the courier needs to travel before reaching the destination. Even if the distance would be shorter than 500 meters, the minimum fee is always 2€.",
      },
      {
        title: "Number of items",
        info: "If the number of items is five or more, an additional 50 cent surcharge is added for each item above and including the fifth item. An extra 'bulk' fee applies for more than 12 items of 1,20€",
      },
      {
        title: "Delivery Time",
        info: "During the Friday rush, 3 - 7 PM, the delivery fee (the total fee including possible surcharges) will be multiplied by 1.2x.",
      },
      {
        title: "Additional rules",
        info: "The delivery fee can never be more than 15€, including possible surcharges. The delivery is free (0€) when the cart value is equal or more than 200€.",
      },
    ],
  },
];
