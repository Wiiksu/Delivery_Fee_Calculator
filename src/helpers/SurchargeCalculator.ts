import { UserFormReturnType, UserFormType } from "./types";

//Free delivery variable
const FREE_DELIVERY_THRESHOLD = 200;

// Small order variable
const SMALL_ORDER_THRESHOLD = 10;

// Delivery distance surcharge variables
const BASE_DELIVERY_FEE = 2;
const BASE_DISTANCE = 1000;
const ADDITIONAL_DISTANCE = 500;
const ADDITIONAL_DISTANCE_SURCHARGE = 1;

// Item number surcharge variables
const ITEM_NUMBER_THRESHOLD = 5;
const ITEM_SURCHARGE = 0.5;
const BULK_THRESHOLD = 12;
const BULK_SURCHARGE = 1.2;

// Rush hour variables
const RUSH_DAY = 5;
const RUSH_HOUR_START = 15;
const RUSH_HOUR_END = 19;
const RUSH_HOUR_MULTIPLIER = 1.2;

// Additional variables
const MAX_SURCHARGE = 15;

/**
 * If the cart value is less than 10€, a small order surcharge is added to the
 * delivery price. The surcharge is the difference between the cart value and
 * 10€.
 * For example if the cart value is 8.90€, the surcharge will be 1.10€.
 * @param cartValue Value of the shopping cart in euros.
 * @returns the small order surcharge if the carts value doesnt exceed 10€
 */
export const underTenSurcharge = (cartValue: number) => {
  return Math.max(SMALL_ORDER_THRESHOLD - cartValue, 0);
};

/**
 * A delivery fee for the first 1000 meters (=1km) is 2€. If the delivery distance is longer than that, 1€ is added for every additional 500 meters that the courier * * needs to travel before reaching the destination. Even if the distance would be shorter than 500 meters, the minimum fee is always 1€.
 * Example 1: If the delivery distance is 1499 meters, the delivery fee is: 2€ base fee + 1€ for the additional 500 m => 3€
 * Example 2: If the delivery distance is 1500 meters, the delivery fee is: 2€ base fee + 1€ for the additional 500 m => 3€
 * Example 3: If the delivery distance is 1501 meters, the delivery fee is: 2€ base fee + 1€ for the first 500 m + 1€ for the second 500 m => 4€
 * @param distanceValue Value of the delivery distance in meters.
 * @returns the delivery distance surcharge
 */
export const distanceSurcharge = (deliveryDistance: number) => {
  if (deliveryDistance <= BASE_DISTANCE) {
    return BASE_DELIVERY_FEE;
  } else {
    deliveryDistance -= BASE_DISTANCE;
    const extraDistance =
      Math.ceil(deliveryDistance / ADDITIONAL_DISTANCE) * ADDITIONAL_DISTANCE;
    const distanceSurchargeMultiplier = extraDistance / ADDITIONAL_DISTANCE;
    const totalDistanceSurcharge =
      ADDITIONAL_DISTANCE_SURCHARGE * distanceSurchargeMultiplier +
      BASE_DELIVERY_FEE;
    return Math.min(totalDistanceSurcharge, MAX_SURCHARGE);
  }
};

/**
 * If the number of items is five or more, an additional 50 cent surcharge is added for each item above and including the fifth item. An extra "bulk" fee applies for more than 12 items of 1,20€
 * Example 1: If the number of items is 4, no extra surcharge
 * Example 2: If the number of items is 5, 50 cents surcharge is added
 * Example 3: If the number of items is 10, 3€ surcharge (6 x 50 cents) is added
 * Example 4: If the number of items is 13, 5,70€ surcharge is added ((9 * 50 cents) + 1,20€)
 * Example 5: If the number of items is 14, 6,20€ surcharge is added ((10 * 50 cents) + 1,20€)
 * @param itemAmount Number of items in the cart.
 * @returns The surcharge based on the amount of items and the possible bulk-surcharge in euros.
 */
export const numberOfItemsSurcharge = (numberOfItems: number) => {
  if (numberOfItems < ITEM_NUMBER_THRESHOLD) {
    return 0;
  } else if (
    numberOfItems >= ITEM_NUMBER_THRESHOLD &&
    numberOfItems <= BULK_THRESHOLD
  ) {
    return (numberOfItems - (ITEM_NUMBER_THRESHOLD - 1)) * ITEM_SURCHARGE;
  } else {
    const totalNumberOfItemsSurcharge =
      (numberOfItems - (ITEM_NUMBER_THRESHOLD - 1)) * ITEM_SURCHARGE +
      BULK_SURCHARGE;
    return Math.min(totalNumberOfItemsSurcharge, MAX_SURCHARGE);
  }
};

/**
 * During the Friday rush, 3 - 7 PM, the delivery fee (the total fee including possible surcharges) will be multiplied by 1.2x. However, the fee still cannot be more
 * than the max (15€). Considering timezone, for simplicity, use UTC as a timezone in backend solutions (so Friday rush is 3 - 7 PM UTC). In frontend solutions, use
 * the timezone of the browser (so Friday rush is 3 - 7 PM in the timezone of the browser).
 * @param orderTime Date/Time when the order is being placed.
 * @param current Sum of possible current surcharges.
 * @returns 0 or conditionally sum of possible current surcharges with the rush hour multiplier.
 */
export const rushHourSurcharge = (
  orderTime: Date,
  currentSurchargeValue: number
) => {
  if (
    orderTime.getDay() === RUSH_DAY &&
    orderTime.getHours() >= RUSH_HOUR_START &&
    orderTime.getHours() < RUSH_HOUR_END
  ) {
    const rushSurcharge = currentSurchargeValue * (RUSH_HOUR_MULTIPLIER - 1);
    return rushSurcharge;
  }
  return 0;
};

/**
 * Calculates smallOrderSurcharge, deliverySurcharge, itemsSurcharge and rushSurcharge based on the values of the formData-param.
 * @param formData that is of type <UserFormType>
 * @returns orderSurchargeDetails-object that is of type <UserFormReturnType>, contains the order surcharge-details.
 */
export function surchargeCalculator(formData: UserFormType) {
  const smallOrderSurcharge = underTenSurcharge(formData.cartValue);
  const deliverySurcharge = distanceSurcharge(formData.deliveryDistance);
  const itemsSurcharge = numberOfItemsSurcharge(formData.numberOfItems);
  const sumOfSurcharges =
    smallOrderSurcharge + deliverySurcharge + itemsSurcharge;
  const rushSurcharge = rushHourSurcharge(
    new Date(formData.orderTime),
    sumOfSurcharges
  );
  const totalSurchargeAmount = sumOfSurcharges + rushSurcharge;

  const orderSurchargeDetails: UserFormReturnType = {
    cartValueSurcharge: smallOrderSurcharge,
    deliveryDistanceSurcharge: deliverySurcharge,
    numberOfItemsSurcharge: itemsSurcharge,
    orderTimeSurcharge: rushSurcharge,
    orderTotal: totalSurchargeAmount,
    freeOrder: false,
  };

  if (totalSurchargeAmount > MAX_SURCHARGE) {
    orderSurchargeDetails.orderTotal = MAX_SURCHARGE;
  }

  if (formData.cartValue >= FREE_DELIVERY_THRESHOLD) {
    // Checks if user is eligible for free delivery
    orderSurchargeDetails.freeOrder = true;
    orderSurchargeDetails.orderTotal = 0;
    return orderSurchargeDetails;
  }
  return orderSurchargeDetails;
}
