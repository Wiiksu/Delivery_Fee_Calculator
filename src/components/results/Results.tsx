import { ResultProps } from "../../helpers/types";
import { MdEmail, MdLocalPhone } from "react-icons/md";
import { LuPackageCheck } from "react-icons/lu";

export default function Results({ orderResults }: ResultProps) {
  const FREE_ORDER = orderResults?.freeOrder;
  const cartValueSurcharge = orderResults?.cartValueSurcharge;
  const deliveryDistanceSurcharge = orderResults?.deliveryDistanceSurcharge;
  const numberOfItemsSurcharge =
    orderResults?.numberOfItemsSurcharge.toFixed(2);
  const orderTimeSurcharge = orderResults?.orderTimeSurcharge.toFixed(2);
  const totalSurcharge = orderResults?.orderTotal;

  return (
    <section className="w-80 rounded bg-white px-6 pt-3 shadow-lg h-fit mt-3 md:mt-0">
      <div className="flex items-center justify-center pt-4 pb-4">
        <LuPackageCheck size={120} color="#a855f7" />
      </div>
      <div className="flex flex-col justify-center items-center gap-2">
        <h4 className="font-semibold">Delivery Business Inc.</h4>
        <p className="text-xs text-center">
          123 Address St. Addressville, Addressland 123456
        </p>
      </div>
      <div className="flex flex-col gap-3 border-b py-6 text-xs">
        <p className="flex justify-between">
          <span className="text-black opacity-45">Receipt Number:</span>
          <span>#12345</span>
        </p>
        <p className="flex justify-between">
          <span className="text-black opacity-45">Order Type:</span>
          <span>Delivery</span>
        </p>
        <p className="flex justify-between">
          <span className="text-black opacity-45">Customer:</span>
          <span>Anonymous</span>
        </p>
      </div>
      <div className="flex flex-col gap-3 pb-6 pt-2 text-xs">
        <table className="w-full text-left">
          <thead>
            <tr className="flex">
              <th className="w-full py-2">Delivery Fee Surcharges</th>
              <th className="min-w-[44px] py-2">Fee</th>
            </tr>
          </thead>
          <tbody>
            <tr className="flex items-center">
              <td className="flex-1 py-1">Cart Value</td>
              <td
                className="min-w-[44px] py-1"
                data-testid="cartValueSurcharge"
              >
                {FREE_ORDER ? 0 : cartValueSurcharge}€
              </td>
            </tr>
            <tr className="flex items-center">
              <td className="flex-1 py-1">Delivery Distance</td>
              <td
                className="min-w-[44px] py-1"
                data-testid="deliveryDistanceSurcharge"
              >
                {FREE_ORDER ? 0 : deliveryDistanceSurcharge}€
              </td>
            </tr>
            <tr className="flex items-center">
              <td className="flex-1 py-1">Item Amount</td>
              <td
                className="min-w-[44px] py-1"
                data-testid="numberOfItemsSurcharge"
              >
                {FREE_ORDER ? 0 : numberOfItemsSurcharge}€
              </td>
            </tr>
            <tr className="flex items-center">
              <td className="flex-1 py-1">Rush Hour</td>
              <td
                className="min-w-[44px] py-1"
                data-testid="orderTimeSurcharge"
              >
                {FREE_ORDER ? 0 : orderTimeSurcharge}€
              </td>
            </tr>
          </tbody>
        </table>
        <table className="w-full text-left">
          <thead>
            <tr className="flex items-center">
              <th className="w-full py-2">Order Total</th>
              <th className="min-w-[44px] py-2">Sum</th>
            </tr>
          </thead>
          <tbody>
            <tr className="flex items-center">
              <td className="flex-1 py-1">Sum of Surcharges</td>
              <td className="min-w-[44px]" data-testid="totalSurcharge">
                {FREE_ORDER ? 0 : totalSurcharge?.toFixed(2)}€
              </td>
            </tr>
            {FREE_ORDER ? (
              <tr className="flex py-1 items-center">
                <td className="flex-1">Details</td>
                <td
                  className="min-w-[44px] text-purple-400 font-semibold"
                  data-testid="freeDelivery"
                >
                  FREE ORDER!
                </td>
              </tr>
            ) : null}
            {totalSurcharge && totalSurcharge >= 15 ? (
              <tr className="flex py-1 items-center">
                <td className="flex-1">Details</td>
                <td
                  className="min-w-[44px] text-purple-400 font-semibold"
                  data-testid="maximumSurcharge"
                >
                  MAXIMUM-SURCHARGE!
                </td>
              </tr>
            ) : null}
          </tbody>
        </table>
        <hr className=" border-b border border-solid" />
        <div className="py-4 px-12 items-start flex flex-col gap-2">
          <p className="flex items-center gap-2">
            <MdEmail fill="#a855f7" size={15} />
            info@deliverybusiness.com
          </p>
          <p className="flex items-center gap-2">
            <MdLocalPhone fill="#a855f7" size={15} />
            +358-XXX-XXXX
          </p>
        </div>
      </div>
    </section>
  );
}
