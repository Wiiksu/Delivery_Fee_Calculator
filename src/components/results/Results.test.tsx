import { expect, it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import Results from "./Results";
import { UserFormReturnType } from "../../helpers/types";

// Testing order-result rendering without special rules
const mockOrderDefault: UserFormReturnType = {
  cartValueSurcharge: 9,
  deliveryDistanceSurcharge: 3,
  numberOfItemsSurcharge: 0.5,
  orderTimeSurcharge: 1,
  orderTotal: 13.5,
  freeOrder: false,
};

// Testing order-result rendering with free-order-rule
const mockOrderFreeDelivery: UserFormReturnType = {
  cartValueSurcharge: 9,
  deliveryDistanceSurcharge: 3,
  numberOfItemsSurcharge: 0.5,
  orderTimeSurcharge: 1,
  orderTotal: 13.5,
  freeOrder: true,
};

// Testing order-result rendering with maximum-surcharge-rule
const mockOrderMaximumSurcharge: UserFormReturnType = {
  cartValueSurcharge: 1,
  deliveryDistanceSurcharge: 12,
  numberOfItemsSurcharge: 2,
  orderTimeSurcharge: 0,
  orderTotal: 15,
  freeOrder: false,
};

describe("<UserForm />", () => {
  it("Should render correctly to the snapshot", () => {
    const { asFragment } = render(<Results orderResults={mockOrderDefault} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should show correctly formatted order details when form values are submitted", () => {
    render(<Results orderResults={mockOrderDefault} />);
    expect(screen.getByTestId("cartValueSurcharge")).toHaveTextContent("9€");
    expect(screen.getByTestId("deliveryDistanceSurcharge")).toHaveTextContent(
      "3€"
    );
    expect(screen.getByTestId("numberOfItemsSurcharge")).toHaveTextContent(
      "0.50€"
    );
    expect(screen.getByTestId("orderTimeSurcharge")).toHaveTextContent("1.00€");
    expect(screen.getByTestId("totalSurcharge")).toHaveTextContent("13.50€");
  });

  it("Should show correct order details when free delivery is activated", () => {
    render(<Results orderResults={mockOrderFreeDelivery} />);
    expect(screen.getByTestId("cartValueSurcharge")).toHaveTextContent("0€");
    expect(screen.getByTestId("deliveryDistanceSurcharge")).toHaveTextContent(
      "0€"
    );
    expect(screen.getByTestId("numberOfItemsSurcharge")).toHaveTextContent(
      "0€"
    );
    expect(screen.getByTestId("orderTimeSurcharge")).toHaveTextContent("0€");
    expect(screen.getByTestId("totalSurcharge")).toHaveTextContent("0€");
    expect(screen.getByTestId("freeDelivery")).toHaveTextContent("FREE ORDER!");
  });

  it("Should show correct order details when maximum surcharge threshold is exceeded", () => {
    render(<Results orderResults={mockOrderMaximumSurcharge} />);
    expect(screen.getByTestId("cartValueSurcharge")).toHaveTextContent("1€");
    expect(screen.getByTestId("deliveryDistanceSurcharge")).toHaveTextContent(
      "12€"
    );
    expect(screen.getByTestId("numberOfItemsSurcharge")).toHaveTextContent(
      "2.00€"
    );
    expect(screen.getByTestId("orderTimeSurcharge")).toHaveTextContent("0.00€");
    expect(screen.getByTestId("totalSurcharge")).toHaveTextContent("15.00€");
    expect(screen.getByTestId("maximumSurcharge")).toHaveTextContent(
      "MAXIMUM-SURCHARGE!"
    );
  });
});
