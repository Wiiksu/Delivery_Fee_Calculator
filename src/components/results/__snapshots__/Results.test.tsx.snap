// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<UserForm /> > Should render correctly to the snapshot 1`] = `
<DocumentFragment>
  <section
    class="w-80 rounded bg-white px-6 pt-3 shadow-lg h-fit mt-3 md:mt-0"
  >
    <div
      class="flex items-center justify-center pt-4 pb-4"
    >
      <svg
        color="#a855f7"
        fill="none"
        height="120"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: #a855f7;"
        viewBox="0 0 24 24"
        width="120"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m16 16 2 2 4-4"
        />
        <path
          d="M21 10V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l2-1.14"
        />
        <path
          d="M16.5 9.4 7.55 4.24"
        />
        <polyline
          points="3.29 7 12 12 20.71 7"
        />
        <line
          x1="12"
          x2="12"
          y1="22"
          y2="12"
        />
      </svg>
    </div>
    <div
      class="flex flex-col justify-center items-center gap-2"
    >
      <h4
        class="font-semibold"
      >
        Delivery Business Inc.
      </h4>
      <p
        class="text-xs text-center"
      >
        123 Address St. Addressville, Addressland 123456
      </p>
    </div>
    <div
      class="flex flex-col gap-3 border-b py-6 text-xs"
    >
      <p
        class="flex justify-between"
      >
        <span
          class="text-black opacity-45"
        >
          Receipt Number:
        </span>
        <span>
          #12345
        </span>
      </p>
      <p
        class="flex justify-between"
      >
        <span
          class="text-black opacity-45"
        >
          Order Type:
        </span>
        <span>
          Delivery
        </span>
      </p>
      <p
        class="flex justify-between"
      >
        <span
          class="text-black opacity-45"
        >
          Customer:
        </span>
        <span>
          Anonymous
        </span>
      </p>
    </div>
    <div
      class="flex flex-col gap-3 pb-6 pt-2 text-xs"
    >
      <table
        class="w-full text-left"
      >
        <thead>
          <tr
            class="flex"
          >
            <th
              class="w-full py-2"
            >
              Delivery Fee Surcharges
            </th>
            <th
              class="min-w-[44px] py-2"
            >
              Fee
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="flex items-center"
          >
            <td
              class="flex-1 py-1"
            >
              Cart Value
            </td>
            <td
              class="min-w-[44px] py-1"
              data-testid="cartValueSurcharge"
            >
              9€
            </td>
          </tr>
          <tr
            class="flex items-center"
          >
            <td
              class="flex-1 py-1"
            >
              Delivery Distance
            </td>
            <td
              class="min-w-[44px] py-1"
              data-testid="deliveryDistanceSurcharge"
            >
              3€
            </td>
          </tr>
          <tr
            class="flex items-center"
          >
            <td
              class="flex-1 py-1"
            >
              Item Amount
            </td>
            <td
              class="min-w-[44px] py-1"
              data-testid="numberOfItemsSurcharge"
            >
              0.50€
            </td>
          </tr>
          <tr
            class="flex items-center"
          >
            <td
              class="flex-1 py-1"
            >
              Rush Hour
            </td>
            <td
              class="min-w-[44px] py-1"
              data-testid="orderTimeSurcharge"
            >
              1.00€
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="w-full text-left"
      >
        <thead>
          <tr
            class="flex items-center"
          >
            <th
              class="w-full py-2"
            >
              Order Total
            </th>
            <th
              class="min-w-[44px] py-2"
            >
              Sum
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="flex items-center"
          >
            <td
              class="flex-1 py-1"
            >
              Sum of Surcharges
            </td>
            <td
              class="min-w-[44px]"
              data-testid="totalSurcharge"
            >
              13.50€
            </td>
          </tr>
        </tbody>
      </table>
      <hr
        class=" border-b border border-solid"
      />
      <div
        class="py-4 px-12 items-start flex flex-col gap-2"
      >
        <p
          class="flex items-center gap-2"
        >
          <svg
            fill="#a855f7"
            height="15"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
          info@deliverybusiness.com
        </p>
        <p
          class="flex items-center gap-2"
        >
          <svg
            fill="#a855f7"
            height="15"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
            />
          </svg>
          +358-XXX-XXXX
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
